### PR TITLE
[Azure Analysis Services] Update set/new operations and fix set bugs

### PR DESCRIPTION
**Pull request by taiwu**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs [hunk 1]
```diff
@@ -26,37 +27,54 @@ namespace Microsoft.Azure.Commands.AnalysisServices
     [Alias("Set-AzureAs")]
     public class SetAzureAnalysisServicesServer : AnalysisServicesCmdletBase
     {
-        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0, Mandatory = true,
+        private const string ParamSetDefault = "Defualt";
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs]: @taiwu typo: Defualt

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs [hunk 2]
```diff
@@ -26,37 +27,54 @@ namespace Microsoft.Azure.Commands.AnalysisServices
...
+        private const string ParamSetDisableBackup = "Disablebackup";
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs]: @taiwu nit: can we change the string to "Disable backup"?

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs [hunk 3]
```diff
@@ -26,37 +27,54 @@ namespace Microsoft.Azure.Commands.AnalysisServices
...
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0, Mandatory = true, 
+            ParameterSetName = ParamSetDefault,
             HelpMessage = "Name of the server.")]
+        [Parameter(ParameterSetName = ParamSetDisableBackup)]
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs]: @taiwu do we want the `ParamSetDisableBackup` parameter set to include all previous parameters _except_ `BackupBlobContainerUri`? If so, we should add that parameter set to the parameters missing it.

Also, please add the attributes that were available for the previous set to this one (_i.e._, `ValueFromPipelineByPropertyName`, `Position`, `Mandatory`, etc.)

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs [hunk 4]
```diff
@@ -26,37 +27,54 @@ namespace Microsoft.Azure.Commands.AnalysisServices
     [Alias("Set-AzureAs")]
     public class SetAzureAnalysisServicesServer : AnalysisServicesCmdletBase
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs]: @taiwu this cmdlet will now need a default parameter set since you have multiple parameter sets. You could also make `DisableBackup` a mandatory parameter to avoid potential issues with PowerShell determining the set to use, but having a default parameter set is the recommended pattern.

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs (hunk 3 above)
Comment (taiwu) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/SetAzureRmAnalysisServicesServer.cs]: Thanks for your advice, the fix is in the new commit.

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.ps1 [hunk 5]
```diff
@@ -228,6 +218,13 @@ function Test-AnalysisServicesServerDisableBackup
 		Assert-AreEqual "Microsoft.AnalysisServices/servers" $serverGetItem.Type
 		Assert-True {$serverGetItem.Id -like "*$resourceGroupName*"}
 		
+		# Update backup container
+	    $backupBlobContainerUriToUpdate = $env:AAS_SECOND_BACKUP_BLOB_CONTAINER_URI
```
Comment (yuchouMS) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.ps1]: tab?

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.ps1 [hunk 6]
```diff
@@ -246,12 +319,13 @@ function Test-AnalysisServicesServerLogExport
...
 		New-AzureRmResourceGroup -Name $resourceGroupName -Location $location
 
-		$serverCreated = New-AzureRmAnalysisServicesServer -ResourceGroupName $resourceGroupName -Name $serverName -Location $location -Sku 'S1' -Administrators 'aztest0@aspaastestloop1.ccsctp.net,aztest1@aspaastestloop1.ccsctp.net'
+		$serverCreated = New-AzureRmAnalysisServicesServer -ResourceGroupName $resourceGroupName -Name $serverName -Location $location -Sku 'S1' -Administrators $env:ASAZURE_TEST_ADMUSERS
 		Assert-True {$serverCreated.ProvisioningState -like "Succeeded"}
 		Assert-True {$serverCreated.State -like "Succeeded"}
 
-        $secpasswd = ConvertTo-SecureString $env.ASAZURE_TESTUSER_PWD -AsPlainText -Force
-		$cred = New-Object System.Management.Automation.PSCredential ('aztest1@aspaastestloop1.ccsctp.net', $secpasswd)
+        $secpasswd = ConvertTo-SecureString $env:ASAZURE_TESTUSER_PWD -AsPlainText -Force
```
Comment (yuchouMS) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.ps1]: ?

File: src/ResourceManager/AnalysisServices/AzureRM.AnalysisServices.psd1 [hunk 7]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '0.4.2'
+ModuleVersion = '1.0.0'
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/AzureRM.AnalysisServices.psd1]: @taiwu please revert this change; we will update the module version as a part of the release

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Azure.AnalysisServices.psd1 [hunk 8]
```diff
@@ -9,7 +9,7 @@
 @{  
   
 # Version number of this module.  
-ModuleVersion = '0.4.2'
+ModuleVersion = '1.0.0'
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Azure.AnalysisServices.psd1]: @taiwu same comment - please revert this

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -35,5 +35,5 @@
 // by using the '*' as shown below:
 
 
-[assembly: AssemblyVersion("0.4.2")]
-[assembly: AssemblyFileVersion("0.4.2")]
+[assembly: AssemblyVersion("1.0.0")]
+[assembly: AssemblyFileVersion("1.0.0")]
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Properties/AssemblyInfo.cs]: @taiwu same comment - please revert this

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/Properties/AssemblyInfo.cs [hunk 10]
```diff
@@ -32,6 +32,6 @@
 
 [assembly: Guid("37af97fd-d603-4d52-a7a6-0885004a57fe")]
 
-[assembly: AssemblyVersion("0.4.2")]
-[assembly: AssemblyFileVersion("0.4.2")]
+[assembly: AssemblyVersion("1.0.0")]
+[assembly: AssemblyFileVersion("1.0.0")]
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/Properties/AssemblyInfo.cs]: @taiwu same comment - please revert this

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.cs [hunk 11]
```diff
@@ -39,6 +39,13 @@ public void TestAnalysisServicesServerScaleUpDown()
         }
 
         [Fact]
+        [Trait(Category.AcceptanceType, Category.LiveOnly)]
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.cs]: @taiwu any reason this test is `LiveOnly`? We want to make sure all tests are `CheckIn` so they are run during the regular CI job

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs [hunk 12]
```diff
@@ -92,7 +97,13 @@ public override void ExecuteCmdlet()
                     }
                 }
 
-                var createdServer = AnalysisServicesClient.CreateOrUpdateServer(ResourceGroupName, Name, Location, Sku, Tag, Administrator);
+                var availableSkus = AnalysisServicesClient.ListSkusForNew();
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs]: @taiwu out of curiosity, why replace the `ValidateSet` with the logic below? Do they not accomplish the same thing? Using a `ValidateSet` was a better experience for users as they were able to tab through the possible values; this extra functionality is now unavailable to the user.

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Properties/AssemblyInfo.cs [hunk 13]
```diff
@@ -34,6 +34,5 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 
-
-[assembly: AssemblyVersion("0.4.2")]
-[assembly: AssemblyFileVersion("0.4.2")]
+[assembly: AssemblyVersion("1.0.0")]
+[assembly: AssemblyFileVersion("1.0.0")]
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Properties/AssemblyInfo.cs]: @taiwu same comment - please revert this

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs [hunk 14]
```diff
@@ -49,19 +50,23 @@ public class NewAnalysisServicesServer : AnalysisServicesCmdletBase
...
-        [ValidateSet("B1", "B2", "S0", "S1", "S2", "S4", "D1")]
         public string Sku { get; set; }
 
         [Parameter(ValueFromPipelineByPropertyName = true, Position = 4, Mandatory = false,
             HelpMessage = "A string,string dictionary of tags associated with this server")]
         [ValidateNotNull]
         public Hashtable Tag { get; set; }
 
-        [Parameter(ValueFromPipelineByPropertyName = true, Position = 4, Mandatory = false,
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 5, Mandatory = false,
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs]: @taiwu this is causing the build to fail as it is considered a breaking change, but it is a false positive since a user would have never been able to supply a value for this position (since it shared the position with `Tag`). You can suppress this exception by copying the line from the `BreakingChangeIssues.csv` file from the build and appending it to the end of [this file](https://github.com/Azure/azure-powershell/blob/preview/tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv).

_Note_: you will need to copy and paste the line from the csv file using a text editor and not Excel since this will cause the quotations to be removed between each item, causing an error to be thrown.

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.cs (hunk 11 above)
Comment (taiwu) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.cs]: The test contains SAS keys, which we do not want to expose them in source code. 
To run tests locally we set keys in enivronment variables, if there are any ways to get the SAS keys, please let me know.

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs (hunk 12 above)
Comment (taiwu) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs]: We have multiple rollouts that has slightly different skus available for customers.
